src/meta/src/barrier/rpc.rs

```diff
@@ -391,3 +399,67 @@ impl StreamRpcManager {
         Ok(())
     }
 }
+
+/// This function is similar to `try_join_all`, but it attempts to collect many error as possible within `error_timeout`.
+async fn try_join_all_with_error_timeout<I, RSP, E, F>(
+    iters: I,
+    error_timeout: Duration,
+) -> (Vec<RSP>, Vec<E>)
+where
+    I: IntoIterator<Item = F>,
+    F: Future<Output = Result<RSP, E>>,
+{
+    let stream = FuturesUnordered::from_iter(iters);
+    pin_mut!(stream);
+    let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
I think we can make the code clearer. It can be changed to have two phase. The first phase is to keep polling the `FuturesUnordered` and, either return when all futures return ok, or break the loop when any future returns err and then enter the second phase. The second phase can be like the following
```
let mut results_err = vec![first_err];
timeout(
    error_timeout,
    async {
        while let Some(result) = stream.next().await {
            if let Err(e) = result {
                results_err.push(e);
            }
        }
    }
).await;
return Err(results_err);
```